pmd-kotlin/pom.xml

```diff
@@ -7,15 +7,29 @@
     <parent>
         <groupId>net.sourceforge.pmd</groupId>
         <artifactId>pmd</artifactId>
-        <version>7.0.0-SNAPSHOT</version>
+        <version>7.0.0-kotlin-SNAPSHOT</version>
```

Comment:
I was surprised to finally find the Kotlin module in this PR... Please fix the history, I appreciate the effort but this is too much of a mess. And why would you change the version number?

Reply:
Thanks for your reply. The mess is because of the rebase. Peter Paul was not sure how to manage it, therefore his question. The version change was to deal with a difficulty we faced. He can better explain that himself and we should find a solution for it with you. Unfortunately he is on vacation right now, but he will try to reply tonight anyway.

Reply:
Ok well I see no particular urgency, if he's on vacation we can let that be until later

Reply:
About the version number: when locally building and trying out pmd `7.0.0-SNAPSHOT` _with_ the Kotlin module from our branch, such as in the pmd-designer and the sonar-pmd plugin, we run into weird issues. We would use a local `pmd-7.0.0-SNAPSHOT`, but it seems when a new one is released in Sonatype, newer than our own local build (e.g. https://oss.sonatype.org/service/local/repositories/snapshots/content/net/sourceforge/pmd/pmd/7.0.0-SNAPSHOT/pmd-7.0.0-20220219.093841-581.pom), builds would use that one (_without_ the Kotlin module), and things would break unexpectedly. By using another version, these version clashes would not occur.

Because we work on the `7.0.x` branch in a fork I thought is would be a good idea to rebase our Kotlin changes on top of the recent `7.0.x` branch, but it seems it does not work as I expected and it messes with the commit history. 

What is the best way to make  a clean PR? Maybe we can make a new branch in our fork on the most recent `7.0.x` branch and cherry-pick our Kotlin commits in that one. 

(and leave out the `7.0.0-kotlin-SNAPSHOT` version and just use that for our local test builds, or find another way to avoid these version clashes)

Reply:
Thanks for replying!

Do you still have a ref to the branch as it was before you rebased (working)? That would be the easiest, just `git reset --hard` to there, and merge the newest `7.0.x` instead of rebasing on it. Then force push.

If you don't then I think checking out the latest `7.0.x`, then checking out the pmd-kotlin module files off the broken branch, then fixing the remaining compilation issues one by one. Cherry-picking commits is probably harder than `git checkout`.

When we merge this PR, the auto-released snapshots will contain your work with the version `7.0.0-SNAPSHOT`. We can't merge a changeset that also changes the version of that snapshot, even if I understand it is helpful locally. Maybe that will fix your issue with sonar though